src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -1,682 +1,588 @@
 package com.eppo.sdk;
 
-import com.eppo.sdk.constants.Constants;
-import com.eppo.sdk.dto.*;
-import com.eppo.sdk.exception.EppoClientIsNotInitializedException;
-import com.eppo.sdk.exception.InvalidInputException;
+import cloud.eppo.ShardUtils;
+import cloud.eppo.rac.Constants;
+import cloud.eppo.rac.dto.*;
+import cloud.eppo.rac.exception.EppoClientIsNotInitializedException;
+import cloud.eppo.rac.exception.InvalidInputException;
 import com.eppo.sdk.helpers.*;
 import com.eppo.sdk.helpers.bandit.BanditEvaluator;
 import com.fasterxml.jackson.databind.JsonNode;
-
-import lombok.extern.slf4j.Slf4j;
-
-import org.ehcache.Cache;
-
 import java.util.*;
 import java.util.stream.Collectors;
+import org.ehcache.Cache;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-@Slf4j
 public class EppoClient {
-    /**
-     * Static Instance
-     */
-    private static EppoClient instance = null;
-    private ConfigurationStore configurationStore;
-
-    private Timer poller;
+  private static final Logger log = LoggerFactory.getLogger(EppoClient.class);
+  private static EppoClient instance = null;
+  private final ConfigurationStore configurationStore;
+  private final Timer poller;
+  private final EppoClientConfig eppoClientConfig;
+
+  private EppoClient(
+      ConfigurationStore configurationStore, Timer poller, EppoClientConfig eppoClientConfig) {
+    this.configurationStore = configurationStore;
+    this.poller = poller;
+    this.eppoClientConfig = eppoClientConfig;
+  }
+
+  /** This function is used to get assignment Value */
+  protected Optional<EppoValue> getAssignmentValue(
+      String subjectKey,
+      String flagKey,
+      EppoAttributes subjectAttributes,
+      Map<String, EppoAttributes> actionsWithAttributes) {
+    Optional<Variation> assignedVariation =
+        getAssignmentVariation(subjectKey, flagKey, subjectAttributes, actionsWithAttributes);
+    return assignedVariation.map(Variation::getTypedValue);
+  }
+
+  /** Returns the assigned variation. */
+  public Optional<Variation> getAssignmentVariation(
+      String subjectKey, String flagKey, EppoAttributes subjectAttributes) {
+    return getAssignmentVariation(subjectKey, flagKey, subjectAttributes, null);
+  }
+
+  /** Returns the assigned variation. */
+  protected Optional<Variation> getAssignmentVariation(
```

Comment:
Presumably, most of this would be in the shared Java code too I reckon? And then each SDK would instantiate a version of the client with defaults that make sense for that environment?

Reply:
yup I think that would make sense to share, I just didnt want to go there quite yet since it's a bigger change